src/react/DataServiceRoleProvider.tsx

```diff
@@ -55,29 +54,25 @@ export const useCurrentAccount = () => {
 };
 
 const DataServiceRoleProvider = ({ children }: { children: JSX.Element }) => {
-  const { accountName } = useParams<{ accountName: string }>();
-  const accountsWithRoles = useAccounts();
   const storedRoleArn = getRoleArnStored();
-  const role = useMemo(() => {
-    if (!storedRoleArn) {
-      const selectedAccount = accountsWithRoles.find(
-        (account) => account.Name === accountName,
-      );
-      const roleArn = selectedAccount?.Roles[0].Arn;
-      return {
-        roleArn: roleArn,
-      };
+  const { account } = useCurrentAccount();
+  const storedAccountID = regexArn.exec(storedRoleArn)?.groups['account_id'];
+
+  const roleArn = useMemo(() => {
+    if (!storedRoleArn || account?.id !== storedAccountID) {
+      // If the account is not the same as the one stored in the localstorage or it's empty, asssume the first Role in the list.
+      return account?.Roles[0].Arn;
     } else {
-      return {
-        roleArn: storedRoleArn,
-      };
+      return storedRoleArn;
     }
-  }, [storedRoleArn, accountName, JSON.stringify(accountsWithRoles)]);
+  }, [storedRoleArn, JSON.stringify(account)]);
+
+  if (roleArn) setRoleArnStored(roleArn);
```

Comment:
I would move this in the useMemo to avoid to run it on each render of `DataServiceRoleProvider`

Reply:
Yes. It makes sense. 
Fixed here: https://github.com/scality/zenko-ui/pull/347/commits/09ec87fd35ea9a98f3ec20db549cf601a41d2a93#diff-a4910c3f27996769a4ea9bbe11ee2438c98732a24475d655f310db91d1114748R65